packages/node_modules/@webex/internal-plugin-device/src/device.js

```diff
@@ -468,38 +468,25 @@ const Device = WebexPlugin.extend({
    * the `services` plugin to confirm if the appropriate service urls are
    * available for device registration.
    *
-   * @param {boolean} [wait=false] - Willing to wait on registration.
    * @returns {Promise<void, Error>}
    */
-  canRegister(wait = false) {
+  canRegister() {
     this.logger.info('device: validating if registration can occur');
 
     // Destructure the services plugin for ease of reference.
     const {services} = this.webex.internal;
 
-    // Validate if the service url exists in the service catalog.
-    if (services.get('wdm')) {
-      return Promise.resolve();
-    }
-
     // If wait is true, wait for the catalog to populate.
-    if (wait) {
-      return services.waitForCatalog(
-        'postauth',
-        this.config.canRegisterWaitDuration
-      )
-        .then(() => (
-          // Validate that the service exists after waiting for the catalog.
-          services.get('wdm') ?
-            Promise.resolve() :
-            Promise.reject(new Error('`wdm` service is not available'))
-        ));
-    }
-
-    // Reject if the service url is not available.
-    return Promise.reject(new Error(
-      '`wdm` service is not currently available'
-    ));
+    return services.waitForCatalog(
+      'postauth',
+      this.config.canRegisterWaitDuration
+    )
+      .then(() => (
+        // Validate that the service exists after waiting for the catalog.
+        services.get('wdm') ?
+          Promise.resolve() :
+          Promise.reject(new Error('`wdm` service is not available'))
```

Comment:
Can we update the error message a bit:

```suggestion
          Promise.reject(new Error('device: cannot register, `wdm` service is not available'))
```

Reply:
implemented.